modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java

```diff
@@ -322,16 +327,43 @@ private JSONArray _getViewsJSONArray(
 							Map<String, Object> fdsFieldProperties =
 								fdsField.getProperties();
 
+							boolean clientExtension = false;
+							String moduleName = null;
+
+							String rendererType = String.valueOf(
+								fdsFieldProperties.get("rendererType"));
+
+							if (rendererType.equals("clientExtension")) {
+								FDSCellRendererCET fdsCellRendererCET =
+									(FDSCellRendererCET)_cetManager.getCET(
+										fragmentEntryLink.getCompanyId(),
+										String.valueOf(
+											fdsFieldProperties.get(
+												"renderer")));
+
+								clientExtension = true;
+								moduleName =
+									"default from " +
+										fdsCellRendererCET.getURL();
+							}
+
 							return JSONUtil.put(
 								"contentRenderer",
 								String.valueOf(
 									fdsFieldProperties.get("renderer"))
+							).put(
+								"contentRendererClientExtension",
+								clientExtension
+							).put(
+								"contentRendererModuleURL", moduleName
```

Comment:
These props don't need to be set if renderer is not a client extension. With this in mind, we can simplify code here, and avoid using two vars (`clientExtension` and `moduleName`) . Something like:
```java
if (Objects.equals(rendererType, "clientExtension")) {
    String rendererType = String.valueOf(
        fdsFieldProperties.get("rendererType"));

    FDSCellRendererCET fdsCellRendererCET =
...
    jsonObject.put(
        "contentRendererClientExtension", true
    }.put(
...
}

return jsonObject;
```

Optionally, you may want to extract parts of code to a private method, just to avoid high amount of line breaks.